azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/message_queue.py

```diff
@@ -0,0 +1,142 @@
+import queue
+import logging
+from threading import Event
+
+logger = logging.getLogger(__name__)
+
+
+class MessageQueue(queue.Queue):
+    def __init__(self, on_enable, on_disable):
+        super(MessageQueue, self).__init__()
+        self._enabled = False
+        self._on_enable = on_enable
+        self._on_disable = on_disable
+
+    def enable(self):
+        self._enabled = True
+        self._on_enable()
+
+    def disable(self):
+        self._enabled = False
+        self._on_disable()
+
+    @property
+    def enabled(self):
+        return self._enabled
+
+    @enabled.setter
+    def enabled(self, v):
+        if v:
+            self.enable()
+        else:
+            self.disable()
+
+
+class MessageQueueManager(object):
```

Comment:
Perhaps MessagingManager might be a better name for what this does (although it doesn't currently, and probably shouldn't, handle sending them).

Think of this as a modular, self-contained piece of the client itself. Thus why it gets the transport (which it needs in order to intelligently enable/disable the c2d/input messages)

Reply:
Agree that this object shouldn't have anything to do with sending messages.  

---
In reply to: [257085321](https://github.com/Azure/azure-iot-sdk-python-preview/pull/49#discussion_r257085321) [](ancestors = 257085321)